coderd/authorize.go

```diff
@@ -81,3 +84,97 @@ func (h *HTTPAuthorizer) Authorize(r *http.Request, action rbac.Action, object r
 	}
 	return true
 }
+
+// checkAuthorization returns if the current API key can use the given
+// permissions, factoring in the current user's roles and the API key scopes.
```

Comment:
It's not clear to me based on this comment why we would have both routes.

What's the purpose of checking authorization, but not including the scope?

Reply:
Agreed. I say delete the old endpoint and use the new one that checks with scopes. The old one apparently let you choose what user you were asking about... which is weird. The new endpoint makes more sense.

Reply:
`checkAuthorization` is mounted at `/api/v2/authorization/can-i` and checks if the current API key can perform those tasks based on the user's roles and the API key's scopes.

`checkUserAuthorization` is mounted at `/api/v2/users/:user/authorization` and does not factor in the current API key at all (other than checking if you're allowed to call the endpoint itself). It checks the permissions of any user based on only their roles. I didn't remove this endpoint as it's used in the frontend in multiple places.

Reply:
Is it used for that purpose though? It seems odd to allow the checking of other user roles, so I'd be surprised if it were used.

Reply:
Thanks to Kyle the user-specific endpoint has been removed